src/ui/control/scroll_zoom_control.js

```diff
@@ -0,0 +1,323 @@
+// @flow
+
+import {extend, bindAll} from '../../util/util.js';
+import {Evented} from '../../util/evented.js';
+import DOM from '../../util/dom.js';
+import window from '../../util/window.js';
+
+import type Map from './../map.js';
+
+const defaultOptions = {
+    closeButton: false,
+    className: ' ',
+    showAlert: true
+};
+
+export type Options = {
+    closeButton?: boolean,
+    className?: string,
+    showAlert?: boolean
+};
+
+/**
+ * A scroll zoom control component.
+ *
+ * @param {Object} [options]
+ * @param {boolean} [options.closeButton=false] If `true`, a close button will appear in the
+ *  top right corner of the scroll zoom control alert.
+ * @param {string} [options.className] Space-separated CSS class names to add to scroll zoom control container.
+ * @example
+ * const scrollZoomControl = new mapboxgl.ScrollZoomControl({closeButton: true})
+ *     .setHTML("<h1>'Use ⌘ + scroll to zoom the map'</h1>")
+ *     .addTo(map);
+ */
+export default class ScrollZoomControl extends Evented {
+    _map: Map;
+    options: Options;
+    _content: HTMLElement;
+    _container: HTMLElement;
+    _closeButton: HTMLElement;
+    _showAlert: any;
+    _classList: Set<string>;
+
+    constructor(options: Options) {
+        super();
+        this.options = extend(Object.create(defaultOptions), options);
+        bindAll(['_update', '_show', '_onClose', 'remove'], this);
+        this._classList = new Set(options && options.className ?
+            options.className.trim().split(/\s+/) : []);
+    }
+
+    /**
+     * Adds the scroll zoom control alert to a map.
+     *
+     * @param {Map} map The Mapbox GL JS map to add the scroll zoom control to.
+     * @returns {ScrollZoomControl} Returns itself to allow for method chaining.
+     * @example
+     * new mapboxgl.ScrollZoomControl().setHTML();
+     * map.addControl(scrollZoomControl);
+     */
+    onAdd(map: Map) {
```

Comment:
`onAdd` has to return "the control's container element" as per the [`IControl` interface](https://docs.mapbox.com/mapbox-gl-js/api/markers/#icontrol). If you open dev tools on the debug page right now, you'll see `Uncaught TypeError: Failed to execute 'appendChild' on 'Node': parameter 1 is not of type 'Node'.`